meetings/2021-03/mar-10.md

```diff
@@ -0,0 +1,846 @@
+
+MM: Prefer a fixed standard page to page size.
+
+YSV: And I guess the next topic is mine as well. So this isn't as much of an issue as having a fixed standard page size, but I want to raise another potential area of compatibility namely when implementations do the rounding. Right now it is also implementation defined, but this could be a place where differences between implementations could become clear. It's not a deal breaker for us, but I would like to hear from JSC what their thoughts are about this.
+
+KM: Right now. We don't do rounding because everything is in wasm page sizes for growing. So in our shared array buffer implementation, we don't have a growable there. So, I don't know. That would probably be something I would have to think deep thoughts about.
+
+SYG: I can add some more color to Yulia's concern. I think one of the things that V8 has talked about internally for implementation strategy. Is that for a small buffer as you might not want to round if you have something that's less than a page or less than n Pages. Maybe you want to just malloc that, and when you resize, you just copy instead of reserving the virtual memory size upfront. That kind of heuristic would become very user observable if that’s implementation defined. For example: when to round up. In that case you can check the max byte length, and if you didn't round, you know, you're in one mode and then if it did round up in another mode and that would have different performance characteristics. To push back on that, I guess is like that was already observable via when you OOM, kind of? And through timing, of course, if the implementation can do that here internally except that if the rounding is not observable the implementation just needs to keep three lengths instead of two.
+
+KM: One option. I don't know if this is reasonable or not. So if we end up picking a fixed page size, we just say like you don't do rounding if it's less than a page and then once you go over a page you do rounding to page sizes. I don't know if that's the right trade-off.
+
+SYG: Between the implementers here let's try to follow up between this meeting and next to come up with the right rounding strategy and page size strategy so that we can you know, maybe say try to make it as deterministic as possible, but still having the trade-offs that we can all live with.
+
+YSV: Yeah, that's fine with me. That sounds good. I was going to suggest that we do a conversation between us as well.
+
+SYG: And I'd certainly want Moddable's input as well.
+
+KM: I'm happy to review the spec for future stages as well.
+
+WH: I'm really impressed by all the changes Shu has made here.
+
+### Conclusion/Resolution
+
+- not stage 3 yet
+- to resolve:
+  - page size rounding thing
+  - wasm
+  - where to put constructors
+
+## Incubation call chartering
+Presenter: Shu-yu Guo (SYG)
+
+- [slides](https://docs.google.com/presentation/d/1oZbRMtnZW78wFUjSY8HXxgsEqeUiXC-2IB2NA_fMV9I/edit)
```

Comment:
This link is not correct. It points to the Promise.race slides

Reply:
Thanks - fixed